### DISCUSSION TITLE
how to save the trained autoML object or all models in it?

### DISCUSSION THREAD

**Discussion by feihongloveworld** [h2oai/h2o-3#15873] [Q&A]
how to save the trained autoML object or all models in it?

**Comment by sebhrusen** [comment]
Hi @feihongloveworld, this is addressed by https://github.com/h2oai/h2o-3/pull/6437
I can't remember why this PR is on standby, we'll try to reactivate it (cc: @tomasfryda )

Right now, you can't save an AutoML object, the only thing you can do is to loop through the trained models and save them individually.

**Reply by tomasfryda** [reply]
I think we put it on hold until the pipelines are merged so we don't create functionality that would be disabled the next release due to new functionality. And also due to other priorities.

I have some prototype implementation but there are missing things - if we load the automl object and do `h2o.get_leaderboard(loaded_automl, extra_colums="ALL")` it can fail since we use some (I think training) frame available during training to calculate the `predict_time_per_row_ms` and we probably don't want to persist that frame along with the automl. We can of course precalculate metrics like this but it could be misleading as it would be relevant only to the training machine (who'd want to know the inference speed on a machine they would not use for inference?).
